src/services/rooms/memberlist/IMemberService.ts

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2024 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { RoomMember } from "../../../models/rooms/RoomMember";
+import { ThreePIDInvite } from "../../../models/rooms/ThreePIDInvite";
+
+export interface IMemberService { 
+    load(): void
+    unload(): void
+    loadMembers(searchQuery: string): Promise<Record<"joined" | "invited", RoomMember[]>>;
+    setOnMemberListUpdated(callback: (reload: boolean) => void): void;
+    setOnPresemceUpdated(callback: (userId: string) => void): void;
+    getThreePIDInvites(): ThreePIDInvite[];
+    shouldShowInvite(): boolean;
+    showPresence(): boolean
```

Comment:
doc comments needed here, what's the diff between load and loadMembers given one implies this concerns only members based on its name

Reply:
Yea some poor naming clashes there, will update and document properly. load/unload were meant as general ViewModel lifecycle functions, `loadMembers` is a relic of the original implementation.